tests/Console/ConfigurationResolverTest.php

```diff
@@ -961,12 +1070,104 @@ public function testResolveConfigFileOverridesDefault()
         $this->assertSame('xml', $resolver->getReporter()->getFormat());
     }
 
+    private function createRuleSetToTestWith(array $rules)
+    {
+        static $testSet = array(
+            '@A' => array(
+                'AA' => true,
+                'AB' => true,
+                'AC' => 'a',
+            ),
+            '@B' => array(
+                'AA' => true,
+            ),
+            '@C' => array(
+                'AA' => false,
+            ),
+            '@D' => array(
+                'AC' => 'b',
+            ),
+        );
+
+        $ruleSet = new RuleSet();
+        $reflection = new \ReflectionObject($ruleSet);
+
+        $setDefinitionsReflectionProperty = $reflection->getProperty('setDefinitions');
+        $setDefinitionsReflectionProperty->setAccessible(true);
+        $setDefinitionsReflectionProperty->setValue($ruleSet, $testSet);
+
+        $setReflectionProperty = $reflection->getProperty('set');
+        $setReflectionProperty->setAccessible(true);
+        $setReflectionProperty->setValue($ruleSet, $rules);
+
+        $resolveSetReflectionMethod = $reflection->getMethod('resolveSet');
+        $resolveSetReflectionMethod->setAccessible(true);
+        $resolveSetReflectionMethod->invoke($ruleSet);
+
+        return $ruleSet;
+    }
+
     private function assertSameRules(array $expected, array $actual, $message = '')
```

Comment:
why can't we just compare 2 arrays ? I don't think we really need that verbose message